File: 396_Rotate_Function/solution.py
class Solution:
    def maxRotateFunction(self, nums: List[int]) -> int:
        t, n, s = 0, len(nums), sum(nums)
        for i, num in enumerate(nums):
            t += i * num
        r = t
        for i in range(n - 1, 0, -1):
            t += s - n * nums[i]
            r = max(r, t)
        return r

File: 396_Rotate_Function/solution.c
int max(int a, int b) { return a > b ? a : b; }
int maxRotateFunction(int* nums, int numsSize) {
    int n = 0, s = 0;
    for (int i = 0; i < numsSize; i++) { n += i * nums[i]; s += nums[i]; }
    int r = n;
    for (int i = numsSize - 1; i > 0; i--) { n += s - numsSize * nums[i]; r = max(r, n); }
    return r;
}